parsers/GE.py

```diff
@@ -3,167 +3,246 @@
 import urllib.parse
 from datetime import datetime, time, timedelta, timezone
 from logging import Logger, getLogger
-from typing import Any
 from zoneinfo import ZoneInfo
 
+import numpy as np
 import pandas
 from requests import Session
 
+from electricitymap.contrib.lib.models.event_lists import (
+    ExchangeList,
+    ProductionBreakdownList,
+)
+from electricitymap.contrib.lib.models.events import ProductionMix
+from electricitymap.contrib.lib.types import ZoneKey
 from parsers.ENTSOE import fetch_exchange as ENTSOE_fetch_exchange
-from parsers.lib import config, validation
+from parsers.lib import config
+from parsers.lib.exceptions import ParserException
 
-MINIMUM_PRODUCTION_THRESHOLD = 10  # MW
+PARSER = "GE.py"
 TIMEZONE = ZoneInfo("Asia/Tbilisi")
+ZONE_KEY = ZoneKey("GE")
+
+MINIMUM_PRODUCTION_THRESHOLD = 10  # MW
 URL = urllib.parse.urlsplit("https://gse.com.ge/apps/gsebackend/rest")
 URL_STRING = URL.geturl()
+SOURCE = URL.netloc
+
+
+def _floor(breakdowns: ProductionBreakdownList, floor: float | int):
+    """Filters production breakdown events whose production sum is lower than a given floor."""
+    filtered_breakdowns = ProductionBreakdownList(breakdowns.logger)
+    for event in breakdowns.events:
+        if event.production is None:
+            continue
+
+        total = sum(v for _mode, v in event.production if v is not None)
+
+        if event.storage is not None:
+            total_storage = sum(v for _mode, v in event.storage if v is not None)
+            total -= total_storage
+
+        if total < floor:
+            filtered_breakdowns.logger.warning(
+                f"Discarded production event for {event.zoneKey} at {event.datetime} due to reported total of {total} MW does not meet {floor} MW floor value."
+            )
+            continue
+
+        filtered_breakdowns.append(
+            zoneKey=event.zoneKey,
+            datetime=event.datetime,
+            production=event.production,
+            storage=event.storage,
+            source=event.source,
+            sourceType=event.sourceType,
+        )
+    return filtered_breakdowns
```

Comment:
This is not needed anymore and can be skipped. We have built and are building out a outlier detection layer in the backend instead.